### PR TITLE
Make wasm hooks importable

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

Make wasm hooks importable by other projects

## Brief Changelog

 
  - Moved ibc hooks into its own module

## Testing and Verifying

all tests pass

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/ibc-hooks/go.mod [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	google.golang.org/genproto v0.0.0-20221024183307-1bc688fe9f3e // indirect
+	google.golang.org/grpc v1.50.1 // indirect
+	google.golang.org/protobuf v1.28.2-0.20220831092852-f930b1dc76e8 // indirect
+	gopkg.in/ini.v1 v1.67.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+	nhooyr.io/websocket v1.8.6 // indirect
+)
+
+replace (
```
Comment (nicolaslara) [inline x/ibc-hooks/go.mod:133]: Open question: do we want the dragonberry replace here or it doesn't matter?

**Comment by nicolaslara** [comment]
Another note here is that the ibc-hooks v0.0.3 is a tag based off this branch. Should we update the tag to be off of main?

File: x/ibc-hooks/go.mod (hunk 1 above)
Comment (ValarDragon) [reply x/ibc-hooks/go.mod:133]: We want it imo

Comment (ValarDragon) [reply x/ibc-hooks/go.mod:133]: Doesn't really matter though, can do in a separate PR

Comment (ValarDragon) [reply x/ibc-hooks/go.mod:134]: I added it in

**Review by ValarDragon — APPROVED** [review]
Fixed conflicts, should be good to merge if it passes CI

**Pull request merged**